src/OSPSuite.SimModelNative/src/FormulaChange.cpp

```diff
@@ -92,6 +92,15 @@ bool FormulaChange::PerformSwitchUpdate(double * y, double time)
 	//---- object to change is not a species - set new formula or value
 	if (_useAsValue)
 	{
+		auto oldFormula = _quantity->GetFormula();
+		if (oldFormula)
```

Comment:
just use `_quantity->AppendUsedVariables(usedIDs, empty)` (it checks that the formula is not null)